src/Service/Modules/StockTriggerLevelsModule.cs

```diff
@@ -0,0 +1,103 @@
+﻿namespace Linn.Stores.Service.Modules
+{
+    using System.Linq;
+
+    using Linn.Common.Authorisation;
+    using Linn.Common.Facade;
+    using Linn.Stores.Domain.LinnApps;
+    using Linn.Stores.Domain.LinnApps.StockLocators;
+    using Linn.Stores.Facade.Services;
+    using Linn.Stores.Resources;
+    using Linn.Stores.Resources.Parts;
+    using Linn.Stores.Resources.RequestResources;
+    using Linn.Stores.Service.Extensions;
+    using Linn.Stores.Service.Models;
+
+    using Nancy;
+    using Nancy.ModelBinding;
+
+    public sealed class StockTriggerLevelsModule : NancyModule
+    {
+        private readonly IStockTriggerLevelsFacadeService stockTriggerLevelsFacadeService;
+
+        private readonly IAuthorisationService authorisationService;
+
+        public StockTriggerLevelsModule(
+            IStockTriggerLevelsFacadeService stockTriggerLevelsFacadeService,
+            IAuthorisationService authorisationService)
+        {
+            this.stockTriggerLevelsFacadeService = stockTriggerLevelsFacadeService;
+            this.authorisationService = authorisationService;
+
+            this.Post("/inventory/stock-trigger-levels/create", _ => this.CreateStockTriggerLevel());
+            this.Get("/inventory/stock-trigger-levels/{id:int}", parameters => this.GetStockTriggerLevel(parameters.id));
+            this.Get("/inventory/stock-trigger-levels", _ => this.SearchStockTriggerLevels());
+            this.Put("/inventory/stock-trigger-levels/{id:int}", parameters => this.UpdateStockTriggerLevel(parameters.id));
+            this.Delete("/inventory/stock-trigger-levels/{id:int}", parameters => this.DeleteStockTriggerLevel(parameters.id));
+        }
+
+        private object SearchStockTriggerLevels()
+        {
+            var resource = this.Bind<SearchRequestResource>();
+
+            var results = this.stockTriggerLevelsFacadeService.Search(resource.SearchTerm);
+
+            return this.Negotiate.WithModel(results).WithMediaRangeModel("text/html", ApplicationSettings.Get)
+                .WithView("Index");
+        }
+
+        private object CreateStockTriggerLevel()
+        {
+            if (!this.authorisationService.HasPermissionFor(
+                    AuthorisedAction.CreateStockTriggerLevel,
+                    this.Context.CurrentUser.GetPrivileges()))
+            {
+                return new UnauthorisedResult<StockTriggerLevel>(
```

Comment:
Your tests are looking for a BadRequest (400) not an Unauthorised (401), so I'm not sure how they're passing. My guess would be the framework doesn't know what to do with UnauthorisedResult<StockTriggerLevel>'s, and so just BadRequest's by default. I think I'd change these to return BadRequest instead of Unauthorised just to be safe